src/backend/core/models.py

```diff
@@ -462,11 +462,20 @@ class Invitation(BaseModel):
         related_name="invitations",
     )
 
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["email", "team"], name="email_and_team_unique_together"
+            )
+        ]
+
     def __str__(self):
         return f"{self.email} invited to {self.team}"
 
     @property
     def is_expired(self):
         """Calculate if invitation is still valid or has expired."""
         validity_duration = timedelta(seconds=settings.INVITATION_VALIDITY_DURATION)
+        if not self.created_at:
+            return False
```

Comment:
I would suggest an early return :
```suggestion
    def is_expired(self):
        """Calculate if invitation is still valid or has expired."""
        if not self.created_at:
            return False
        validity_duration = timedelta(seconds=settings.INVITATION_VALIDITY_DURATION)
```

Reply:
Why did you choose `False` over `None`?

Reply:
Well, I'm not even sure if it could happen in a real situation, I wrote that when testing `is expired` before saving object. I'm still leaving it, in case it occurs in another situation.

No objection whatsoever to return "None" instead of "False".